aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/io/TlsContextNative.kt

```diff
@@ -5,15 +5,119 @@
 
 package aws.sdk.kotlin.crt.io
 
-import aws.sdk.kotlin.crt.Closeable
+import aws.sdk.kotlin.crt.*
+import aws.sdk.kotlin.crt.Allocator
+import aws.sdk.kotlin.crt.awsAssertOpSuccess
+import aws.sdk.kotlin.crt.util.asAwsByteCursor
+import aws.sdk.kotlin.crt.util.free
+import aws.sdk.kotlin.crt.util.toAwsString
+import kotlinx.cinterop.*
+import libcrt.*
+
+@OptIn(ExperimentalForeignApi::class)
+public actual class TlsContext actual constructor(options: TlsContextOptions?) : CrtResource<aws_tls_ctx>(), Closeable {
+    private val ctx: CPointer<aws_tls_ctx>
+    private val tlsCtxOpts: aws_tls_ctx_options = Allocator.Default.alloc()
 
-public actual class TlsContext actual constructor(options: TlsContextOptions?) : Closeable {
     public actual companion object {}
 
+    @OptIn(ExperimentalForeignApi::class)
+    override val ptr: CPointer<aws_tls_ctx>
+        get() = ctx
+
+    init {
+        aws_tls_ctx_options_init_default_client(tlsCtxOpts.ptr, Allocator.Default)
+        val kopts = options ?: TlsContextOptions.defaultClient()
+
+        try {
+            // Certificate file or path will cause an init which overrides other fields, so parse those first
+            if (kopts.certificate != null && kopts.privateKey != null) {
+                initClientMtls(kopts.certificate, kopts.privateKey)
+            } else if (kopts.certificatePath != null && kopts.privateKeyPath != null) {
+                initClientMtlsFromPath(kopts.certificatePath, kopts.privateKeyPath)
+            }
+
+            if (kopts.caRoot != null) {
+                overrideDefaultTrustStore(kopts.caRoot)
+            } else if (kopts.caFile != null && kopts.caDir != null) {
+                overrideDefaultTrustStoreFromPath(kopts.caFile, kopts.caDir)
+            }
+
+            tlsCtxOpts.minimum_tls_version = kopts.minTlsVersion.value.convert()
+            tlsCtxOpts.cipher_pref = kopts.tlsCipherPreference.value.convert()
+            tlsCtxOpts.verify_peer = kopts.verifyPeer
+
+            if (kopts.alpn.isNotBlank()) {
+                awsAssertOpSuccess(aws_tls_ctx_options_set_alpn_list(tlsCtxOpts.ptr, kopts.alpn)) {
+                    "aws_tls_ctx_options_set_alpn_list() failed"
```

Comment:
**style**: I think we can just have `"aws_tls_ctx_options_set_alpn_list()"` or more generally `"crt_fn_name()"` for these sort of assertions. "failed" adds nothing more since it's going to throw an exception anyway.

Reply:
Are these exceptions going to be shown to consumers? If so, shouldn't they be a little more informative?